lib/src/git.rs

```diff
@@ -1275,50 +1275,30 @@ pub fn push_updates(
     updates: &[GitRefUpdate],
     callbacks: RemoteCallbacks<'_>,
 ) -> Result<(), GitPushError> {
-    let mut temp_refs = vec![];
     let mut qualified_remote_refs = vec![];
     let mut refspecs = vec![];
     for update in updates {
         qualified_remote_refs.push(update.qualified_name.as_str());
         if let Some(new_target) = &update.new_target {
-            // Create a temporary ref to work around https://github.com/libgit2/libgit2/issues/3178
-            let temp_ref_name = format!("refs/jj/git-push/{}", new_target.hex());
-            temp_refs.push(git_repo.reference(
-                &temp_ref_name,
-                git2::Oid::from_bytes(new_target.as_bytes()).unwrap(),
-                true,
-                "temporary reference for git push",
-            )?);
             refspecs.push(format!(
                 "{}{}:{}",
                 (if update.force { "+" } else { "" }),
-                temp_ref_name,
+                git2::Oid::from_bytes(new_target.as_bytes()).unwrap(),
```

Comment:
nit: `new_target.hex()` also works

Reply:
`new_target.hex()` returns a String, and this function needs `&[u8]`. I supposed we could implement `Into<git2::Oid>` for CommitId. I might keep it as is for now.

Reply:
I mean `git2::Oid` can be removed at all. It's just there because `Display` is implemented as a hex formatting function.

Reply:
Ah, of course. It's being converted to a string.